assignment3/Dictionary.java

```diff
@@ -1,35 +1,50 @@
 package com.katermar.assignment3;
 
+import java.util.Arrays;
 import java.util.HashSet;
 
 /**
  * Created by katermar on 6/8/2017.
  */
 public class Dictionary {
-    private HashSet<String> dictionary = new HashSet<>();
-    private HashSet<String> prefixes = new HashSet<>();
+    private final HashSet<String> dictionary;
+    private final HashSet<String> prefixes;
 
-    public boolean isPrefix(String prefix){
-      return prefixes.contains(prefix);
+    public Dictionary(HashSet<String> dictionary, HashSet<String> prefixes) {
+        this.dictionary = dictionary;
+        this.prefixes = prefixes;
     }
 
-    public boolean isWord(String word) {
-        return dictionary.contains(word);
+    public Dictionary(String[] dictionary, String[] prefixes) {
+        this.dictionary = new HashSet<>(Arrays.asList(dictionary));
+        this.prefixes = new HashSet<>(Arrays.asList(prefixes));
     }
 
-    public HashSet<String> getDictionary() {
-        return dictionary;
+    public Dictionary(String[] dictionary) {
+        this.dictionary = new HashSet<>(Arrays.asList(dictionary));
+        this.prefixes = new HashSet<>();
+        for (String word : dictionary) {
+            for (int i = 0; i < word.length(); i++) {
+                prefixes.add(word.substring(0, i));
+            }
+        }
     }
 
-    public void setDictionary(HashSet<String> dictionary) {
+    public Dictionary(HashSet<String> dictionary) {

```

Comment:
Just FYI comment. Feel free to not the code.

It's better to use interface (`Set`) as parameter type rather than concrete type (`HashSet`). It gives much more flexibility for the users of the class - they could prefer using `TreeSet` or something similar.
To get the performance guarantees of a `HashSet` in your code (and to protect from a malicious user that might change contents of the dictionary set without your knowledge) you would just copy it in constructor.